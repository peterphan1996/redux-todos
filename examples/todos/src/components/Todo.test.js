import React from 'react';
import { shallow } from 'enzyme';

import Todo from './Todo';

const setup = setupProps => {
  const defaultProps = {
    text: 'test',
    completed: false,
    onClick: jest.fn(),
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <Todo text={props.text} completed={props.completed} onClick={props.onClick} />
  );

  return {
    props,
    wrapper,
  };
};

describe('Todo', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('have a line-through when prop passed completed', () => {
    const { wrapper } = setup({ completed: true });

    expect(wrapper.find('li').prop('style')).toEqual({
      textDecoration: 'line-through'
    });
  });

  test('calls onClick once when clicked', () => {
    const { props, wrapper } = setup();
    wrapper.find('li').simulate('click');
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});