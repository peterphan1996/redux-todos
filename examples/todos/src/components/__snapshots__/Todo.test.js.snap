// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders without crashing 1`] = `
<li
  onClick={[MockFunction]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  test
</li>
`;
